taxcalc/tests/test_policy.py

```diff
@@ -837,77 +782,15 @@ def test_description_punctuation(tests_path):
-    min_max_list = ['min', 'max']
-    warn_stop_list = ['warn', 'stop']
-    json_range_params = set()
-    parameters = set(policy._vals.keys())
-    for pname, param in policy._vals.items():
-        assert isinstance(param, dict)
-        if param['value_type'] == 'string':
-            continue  # because string parameters have no invalid_* keys
-        prange = param.get('valid_values', None)
-        if prange:
-            json_range_params.add(pname)
-            oor_action = param['invalid_action']
-            assert oor_action in warn_stop_list
-            range_items = prange.items()
-            assert len(range_items) == 2
-            for vop, vval in range_items:
-                assert vop in min_max_list
-                if isinstance(vval, str):
-                    if vval == 'default':
-                        if vop != 'min' or oor_action != 'warn':
-                            msg = 'USES DEFAULT FOR min OR FOR error'
-                            assert pname == msg
-                        continue
-                    else:
-                        vval_ = '_' + vval
-                        if vval_ in parameters:
-                            if vop == 'min':
-                                extra_msg = param['invalid_minmsg']
-                            if vop == 'max':
-                                extra_msg = param['invalid_maxmsg']
-                            assert vval in extra_msg
-                        else:
-                            assert vval == 'ILLEGAL RANGE STRING VALUE'
-                else:  # if vval is not a str
-                    if isinstance(vval, int):
-                        continue
-                    elif isinstance(vval, float):
-                        continue
-                    elif isinstance(vval, bool):
-                        continue
-                    else:
-                        assert vval == 'ILLEGAL RANGE NUMERIC VALUE'
-    # compare contents of c_l_p.json parameters and json_range_params
-    unmatched = parameters ^ json_range_params
-    if unmatched:
-        assert unmatched == 'UNMATCHED RANGE PARAMETERS'
-    # check all current-law-policy parameters for range validity
-    clp = Policy()
-    clp._validate_values(parameters)
-    # eventually activate: assert not clp.parameter_warnings
-    ctc_c_warning = 'CTC_c was redefined in release 1.0.0\n'
-    assert clp.parameter_warnings == ctc_c_warning
-    assert not clp.parameter_errors
-
-
 def test_indexing_rates_for_update():
     """
     Check private _indexing_rates_for_update method.
     """
     pol = Policy()
-    wgrates = pol._indexing_rates_for_update('_SS_Earnings_c', 2017, 10)
-    pirates = pol._indexing_rates_for_update('_II_em', 2017, 10)
-    assert len(wgrates) == len(pirates)
+    wgrates = pol.get_index_rate('_SS_Earnings_c', 2017)
+    pirates = pol.get_index_rate('_II_em', 2017)
+    assert isinstance(wgrates, np.float64)
+    assert isinstance(pirates, np.float64)
```

Comment:
`Parameters._indexing_rates_for_update` is no longer used. A public method for getting the index rate of a paramter at a given year is used instead: [`Parameters.get_index_rate`](https://github.com/hdoupe/Tax-Calculator/blob/use-paramtools/taxcalc/parameters.py#L315-L327).